### ISSUE TITLE
Refactor the menu

### ISSUE THREAD

**Issue by codec-abc** [RICCIARDI-Adrien/Strage#3]
I mean, [look at it](https://github.com/RICCIARDI-Adrien/Strage/blob/master/Sources/Menu.cpp). You know you want to refactor it. It uses while loop everywhere, and the stack is used to maintain the state which makes the code hard to follow. Also, it uses number to communicate what happen to the caller which is quite error prone (enums would be better).

**Comment by dcarlus** [comment]
Game state pattern?
